src/aiida/cmdline/groups/verdi.py

```diff
@@ -79,6 +80,30 @@ def __init__(self, *args, **kwargs):
             self.obj = LazyVerdiObjAttributeDict(self)
 
 
+class VerdiCommand(click.Command):
+    """Custom command implementation to change the logic how the deprecation warning is handled. In the parent class the
+    warning is printed only on invocation and after the parsing of the arguments. In this class we print the warning in
+    addition when the help page is printed and change the behavior on invocation. It is printed on invocation before the
+    arguments are parsed. That allows a printing of the deprecation warning with the first prompt requesting missing
+    arguments."""
```

Comment:
```suggestion
class VerdiCommand(click.Command):
    """Custom command implementation to customize the logic of printing deprecation messages.
    
    If a command is deprecated, the :class:`click.Command` adds a deprecation marker in the short help and the full
    help text, and prints a deprecation warning when the command is invoked. The problem is that the deprecation warning
    is printed after the prompting for parameters, which for interactive commands mean the deprecation warning comes too
    late, when the user has already provided all prompts.
    
    Here, the :meth:`click.Command.parse_args` method is overridden, which is called before the interactive options
    start to prompt, such that the deprecation warning can be printed. The :meth:`click.Command.invoke` method is also
    overridden in order to skip the printing of the deprecation message handled by ``click`` as that would result in
    the deprecation message being printed twice.
    """
```

Reply:
Reworded it and added the reason why `invoke` is also overridden which is not immediately clear